packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts

```diff
@@ -216,19 +217,57 @@ export class ImportCard extends LitElement {
     dropControl.on('drop', ({files}: any) => this.onUpload(files));
   }
 
-  async onUpload(files) {
-    if (!files) {
-      /// The user canceled the previous upload
-      return;
+  async onUpload(fileMap: Map<string, File>) {
+    const modelViewer = getModelViewer();
+    let rootPath: string;
+    for (const [path, file] of fileMap) {
+      const filename = file.name.toLowerCase();
+      if (filename.match(/\.(gltf|glb)$/)) {
```

Comment:
Nit: It looks like maybe this loop is just finding the .gltf/.glb file and then doing some stuff with it? If so I think you can break at the bottom of _if_ statement yeah?

Reply:
NM braces are difficult to follow in this view